builds/azure-pipelines/template-steps-build-test.yml

```diff
@@ -60,6 +65,11 @@ steps:
   displayName: Set logging level
   condition: and(succeeded(), ne(variables['AFSQLEXT_TEST_LOGLEVEL'], ''))
 
+- task: npmAuthenticate@0
+  displayName: 'npm Authenticate for project'
+  inputs:
+  workingFile: $(Build.SourcesDirectory)/.npmrc
+  
 - script: |
     npm install
```

Comment:
What about this? (and other ones below)

Reply:
Oh right, nevermind. Local installs pick up the local config. It's only the global ones that need to specify the config directly.

Reply:
added this las week and forgot to clean up, I just cleaned up and added userconfig for the below ones.

Reply:
What I understood is, when using as a per project config, it needs to be a sibling of the package.json but when you're specifying it as userconfig, I think you can give the path to the file. But am not 100% sure either. Do you think if I should be adding a .npmrc file here as well?

Reply:
So the issue we have is that we have multiple projects to build (e.g. samples-js and test-js). So in order for it to pick up the npmrc directly we'd have to have one in each of those folders - it doesn't seem to contain any logic to automatically look in parent folders unfortunately.

What you have seems fine - just setting it at the user level will cause it to use that instead. But yes, you will need to add an npmrc file in order for this to successfully build.